internal/scan-manager/scanners/veracode/src/main/java/org/wso2/security/tools/scanmanager/scanners/veracode/handler/VeracodeResultProcessor.java

```diff
@@ -0,0 +1,281 @@
+     */
+    public static String getFullErrorMessage(Throwable e) {
+        if (e.getCause() == null) {
+            return e.getMessage();
+        }
+        return e.getMessage() + "\n\nCaused by: " + getFullErrorMessage(e.getCause());
+    }
+
+    /**
+     * Get the required attribute from an XML sttring.
+     *
+     * @param result    XML of the Veracode response
+     * @param xPath     to get the required attribute
+     * @param attribute tag name of the required attribute
+     * @return attribute value
+     * @throws SAXException
+     * @throws ParserConfigurationException
+     * @throws XPathExpressionException
+     * @throws IOException
+     */
+    private static String getElementByXMLResult(String result, String xPath, String attribute) throws
+            XPathExpressionException, IOException, SAXException, ParserConfigurationException {
+        String status = null;
+        XPathFactory xPathfactory = XPathFactory.newInstance();
+        XPath xpath = xPathfactory.newXPath();
+        XPathExpression expr;
+        NodeList nl;
+
+        Document doc = convertStringToDocument(result);
+
+        SimpleNamespaceContext namespaces = new SimpleNamespaceContext();
+        namespaces.bindNamespaceUri(VeracodeScannerConstants.VERACODE, YAMLConfigurationReader.getInstance()
+                .getConfigProperty(VeracodeScannerConstants.NAMESPACE));
+        xpath.setNamespaceContext(namespaces);
+        expr = xpath.compile(xPath);
+
+        nl = (NodeList) expr.evaluate(doc, XPathConstants.NODESET);
+
+        for (int i = 0; i < nl.getLength(); i++) {
+            Node currentItem = nl.item(i);
+            status = currentItem.getAttributes().getNamedItem(attribute).getNodeValue();
+        }
+        return status;
+    }
+
+    /**
+     * Convert a given XML to XML Document.
+     *
+     * @param xmlStr XML string that needs to convert
+     * @return converted XML Document
+     * @throws ParserConfigurationException
+     * @throws IOException
+     * @throws SAXException
+     */
+    private static Document convertStringToDocument(String xmlStr) throws ParserConfigurationException, IOException,
+            SAXException {
+        DocumentBuilderFactory factory = FileUtil.getSecuredDocumentBuilderFactory();
+        DocumentBuilder builder = factory.newDocumentBuilder();
+
+        return builder.parse(new InputSource(new StringReader(xmlStr)));
+    }
+
+    /**
+     * Convert the Veracode scan status to generic scan status for a given Veracode response result.
+     *
+     * @param result Veracode API response
+     * @return scan status
+     * @throws SAXException
+     * @throws ParserConfigurationException
+     * @throws XPathExpressionException
+     * @throws IOException
+     */
+    public static ScanStatus getScanStatus(String result) throws SAXException, ParserConfigurationException,
+            XPathExpressionException, IOException {
+        ScanStatus scanStatus = ScanStatus.COMPLETED;
```

Comment:
Default value should not be `COMPLETED`. Let's have an state called `UNKNOWN` for such cases.